go/vt/vtgate/scatter_conn.go

```diff
@@ -386,6 +392,15 @@ func (stc *ScatterConn) Commit(context context.Context, session *SafeSession) (e
 
 // Rollback rolls back the current transaction. There are no retries on this operation.
 func (stc *ScatterConn) Rollback(context context.Context, session *SafeSession) (err error) {
+	if session == nil {
```

Comment:
For rollback, I think it's ok to not be strict at all. So, it's ok not to check anything. This is because clients often do redundant rollbacks 'for good measure'.
Errors here have mostly ended up alarming people unnecessarily.